src/esp/bindings/CoreBindings.cpp

```diff
@@ -14,25 +14,177 @@ namespace esp {
+      .def("get_value_types", &Configuration::getValueTypes,
+           R"(Returns a dictionary where the keys are the names of the values
+           this configuration holds and the values are the types of these values.)")
+      .def(
+          "get_subconfig_keys", &Configuration::getSubconfigKeys,
+          R"(Retrieves a list of the keys of this configuration's subconfigurations)")
+
+      .def("get_subconfig", &Configuration::editSubconfig,
+           pybind11::return_value_policy::reference,
+           R"(Get the subconfiguration with the given name.)", "name"_a)
+      .def("get_subconfig_copy", &Configuration::getSubconfigCopy,
+           pybind11::return_value_policy::reference,
+           R"(Get a copy of the subconfiguration with the given name.)",
+           "name"_a)
+      .def("save_subconfig", &Configuration::setSubconfigPtr,
+           R"(Save a subconfiguration with the given name.)", "name"_a,
+           "subconfig"_a)
+      .def("set", [](Configuration& self, const std::string& key,
+                     const std::string& val) { self.set(key, val); })
+      .def("set", [](Configuration& self, const std::string& key,
+                     const char* val) { self.set(key, val); })
+      .def("set", [](Configuration& self, const std::string& key,
+                     const int val) { self.set(key, val); })
+      .def("set", [](Configuration& self, const std::string& key,
+                     const double val) { self.set(key, val); })
+      .def("set", [](Configuration& self, const std::string& key,
+                     const bool val) { self.set(key, val); })
+      .def("set", [](Configuration& self, const std::string& key,
+                     const Magnum::Quaternion& val) { self.set(key, val); })
+      .def("set", [](Configuration& self, const std::string& key,
+                     const Magnum::Vector3& val) { self.set(key, val); })
+
+      .def("set_string", [](Configuration& self, const std::string& key,
+                            const std::string& val) { self.set(key, val); })
+      .def("set_int", [](Configuration& self, const std::string& key,
+                         const int val) { self.set(key, val); })
+      .def("set_float", [](Configuration& self, const std::string& key,
+                           const double val) { self.set(key, val); })
+      .def("set_bool", [](Configuration& self, const std::string& key,
+                          const bool val) { self.set(key, val); })
+      .def("set_quat",
+           [](Configuration& self, const std::string& key,
+              const Magnum::Quaternion& val) { self.set(key, val); })
+      .def("set_vec3", [](Configuration& self, const std::string& key,
+                          const Magnum::Vector3& val) { self.set(key, val); })
+      .def("set_rad", [](Configuration& self, const std::string& key,
+                         const Magnum::Rad& val) { self.set(key, val); })
```

Comment:
Back-compat with the original "set" name - was trying to be as compatible as possible, since I wasn't sure how much the original "set" verbiage was expected.  
My preferences were, when I pushed the most recent commit, to have the type explicitly named in all the accessors, so there as no confusion or mistakes.  While this is exceedingly verbose, this was a first pass at being as safe as possible.  Since this is only pertinent for configurations that the user is customizing and creating on the fly, either in python or in JSON files (not the built ins, which will always be accessed directly as properties), I wanted to provide as many guardrails as possible.